pro/src/data_access/ApiRecipeDataAccessObject.java

```diff
@@ -1,68 +1,83 @@
 package data_access;
 
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
 import okhttp3.OkHttpClient;
 import okhttp3.Request;
 import okhttp3.Response;
 import org.json.JSONObject;
+import service.search.use_case.SearchDataAccessInterface;
 
-import java.io.IOException;
-import java.util.HashMap;
-
-
-public class ApiRecipeDataAccessObject {
+public class ApiRecipeDataAccessObject implements SearchDataAccessInterface {
     private final String keyword;
 
-    public ApiRecipeDataAccessObject(String keyword){
+    public ApiRecipeDataAccessObject(String keyword) {
```

Comment:
To better apply the design pattern, we can use the builder pattern and factory pattern to construct a Recipe given that a Recipe is complex enough.

Reply:
Do you mean we need to create "RecipeFactory" and its interface in the entity folder?

Reply:
Yes